### PR TITLE
Update compatibility for temporary tables in dev

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/4935
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-create-table.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ mysql> DESC t1;
 
 ## MySQL 兼容性
 
-* TiDB 不支持临时表，对于 `CREATE TEMPORARY TABLE` 语法，会忽略 `TEMPORARY` 关键字。
+* TiDB 不支持临时表。如果  [`tidb_enable_noop_functions = 0`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) 时，`CREATE TEMPORARY TABLE` 语法会报错。如果`tidb_enable_noop_functions = 1` 时， TiDB 只会忽略 `TEMPORARY` 关键字。
```
Comment (TomShawn) [inline sql-statements/sql-statement-create-table.md:184]:
```suggestion
* TiDB 不支持临时表。如果 [`tidb_enable_noop_functions = 0`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) ，执行 `CREATE TEMPORARY TABLE` 语法会报错。如果 `tidb_enable_noop_functions = 1`， TiDB 会忽略 `TEMPORARY` 关键字。
```

**Review by TomShawn — COMMENTED** [review]
Rest LGTM

File: sql-statements/sql-statement-create-table.md [hunk 2]
```diff
@@ -181,7 +181,7 @@ mysql> DESC t1;
 
 ## MySQL 兼容性
 
-* TiDB 不支持临时表，对于 `CREATE TEMPORARY TABLE` 语法，会忽略 `TEMPORARY` 关键字。
+* TiDB 不支持临时表。如果 [`tidb_enable_noop_functions = 0`](/system-variables.md#tidb_enable_noop_functions-new-in-v40) ，执行 `CREATE TEMPORARY TABLE` 语法会报错。如果 `tidb_enable_noop_functions = 1`， TiDB 会忽略 `TEMPORARY` 关键字。
```
Comment (TomShawn) [inline sql-statements/sql-statement-create-table.md:184]:
```suggestion
* TiDB 不支持临时表。如果 [`tidb_enable_noop_functions = 0`](/system-variables.md#tidb_enable_noop_functions-从-v40-版本开始引入) ，执行 `CREATE TEMPORARY TABLE` 语法会报错。如果 `tidb_enable_noop_functions = 1`， TiDB 会忽略 `TEMPORARY` 关键字。
```

**Comment by TomShawn** [comment]
LGTM

**Review by bb7133 — COMMENTED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]

**Comment by TomShawn** [comment]
Follow-up: After 5.0 GA release, cherry-pick this PR to release-5.0 branch and add "since v5.0".

**Comment by yikeke** [comment]
You can cherry-pick it now and make it ready for merge. @TomShawn

**Comment by TomShawn** [comment]
> You can cherry-pick it now and make it ready for merge on 5.0 branch. @TomShawn

Updated in https://github.com/pingcap/docs-cn/pull/5758.

**Pull request merged**